state/upgrades.go

```diff
@@ -1388,3 +1388,24 @@ func addRelationStatus(st *State) error {
 	})
 	return errors.Annotate(err, "adding relation status")
 }
+
+// CopyMongoSpaceToHASpaceConfig copies the Mongo space name from
+// ControllerInfo to the HA space name in ControllerConfig.
+func CopyMongoSpaceToHASpaceConfig(st *State) error {
```

Comment:
I am starting to believe that you have a knack for writing exceptionally clear tests.